Signal/src/ViewControllers/HomeViewController.m

```diff
@@ -931,10 +931,12 @@ - (void)yapDatabaseModifiedExternally:(NSNotification *)notification
 
     DDLogVerbose(@"%@ %s", self.logTag, __PRETTY_FUNCTION__);
 
-    // External database modifications can't be converted into incremental updates,
-    // so rebuild everything.  This is expensive and usually isn't necessary, but
-    // there's no alternative.
-    [self resetMappings];
+    if (self.shouldObserveDBModifications) {
```

Comment:
Small, safe optimization: we don't need to do this if we're not observing db modifications since we'll do it when we resume.

Reply:
Makes sense -can you leave a similar comment in the code?

Reply:
Done.